drivers/gles3/rasterizer_canvas_gles3.cpp

```diff
@@ -1141,7 +1141,17 @@ void RasterizerCanvasGLES3::canvas_render_items(Item *p_item_list, int p_z, cons
 			if (current_clip) {
 
 				glEnable(GL_SCISSOR_TEST);
-				glScissor(current_clip->final_clip_rect.position.x, (rt_size.height - (current_clip->final_clip_rect.position.y + current_clip->final_clip_rect.size.height)), current_clip->final_clip_rect.size.width, current_clip->final_clip_rect.size.height);
+
+				real_t clip_y = (rt_size.height - (current_clip->final_clip_rect.position.y + current_clip->final_clip_rect.size.height));
+				real_t clip_height = current_clip->final_clip_rect.size.height;
+
+				// If the render target is flipped, we also need to flip the scissor rectangle.
+				if (storage->frame.current_rt && storage->frame.current_rt->flags[RasterizerStorage::RENDER_TARGET_VFLIP]) {
+					real_t half_height = rt_size.height * 0.5;
+					clip_y = -(clip_y - half_height) + half_height - clip_height;
```

Comment:
this is confusing too:
`-(clip_y - half_height) + half_height - clip_height` is the same than
`- clip_y + 2*half_height - clip_height;` again the same than `- clip_y + rt_size.height - clip_height;`
`- rt_size.height + current_clip->final_clip_rect.position.y + current_clip->final_clip_rect.size.height +  rt_size.height - clip_height`
`current_clip->final_clip_rect.position.y + current_clip->final_clip_rect.size.height - clip_height`
`current_clip->final_clip_rect.position.y`

so this line should just be:
clip_y = current_clip->final_clip_rect.position.y which also make a lot of sense if you think about it.
so there is no need for complex calculations!

Sorry for being such a dick at the moment ;) I'm still proposing to merge my version. It's less confusing...
It took me quiet some time to find the issue. If it also took you some time it might be annoying to close your pr. So if you want to, you can copy my changes and we merge yours. You also need to rebase though since a unnecessary merge commit is not desired.